### ISSUE TITLE
[FEATURE REQ] Nullable reference types support in Azure.Messaging.EventHubs

### ISSUE THREAD

**Issue by yuretz** [Azure/azure-sdk-for-net#33004]
### Library name

Azure.Messaging.EventHubs

### Please describe the feature.

Please consider supporting nullable reference type annotations in _Azure.Messaging.EventHubs_. It would make it much easier to use the public APIs and types provided in the correct way, because sometimes it is so not obvious whether something could be null or not, and the documentation is silent on that topic.

At the very least, it would be very helpful to clearly document it somewhere, whether the library supports nullable reference types or not, in order to not to create any false expectations.

**Comment by jsquire** [comment]
Hi @yuretz.  Thank you for reaching out and we regret that you're experiencing difficulties.   Nullable reference annotations are something that we continue to evaluate for Azure SDK packages as the language and tooling evolves, but not something that we'll be enabling in the short term.   There are several reasons for this, the most important being:

- The Azure SDK strives to have a consistent set of patterns and feel across different libraries.   As such, nullable reference annotations are not something that would be applied to individual libraries; it would be a decision that we need to apply to all of our current generation packages.

- For most of our libraries, the service definitions do not adequately document which fields may be null.   We continue to work with our service partners to improve these definitions, but it's a large effort that we expect to continue for some time. 

- Tooling still doesn't provide a thorough understanding of what breaking changes would be introduced by enabling nullable reference annotations.  Avoiding breaking changes is one of the most important core tenants of the current Azure SDK generation; until we have confidence that we can prove that nullable reference annotations are safe for the public APIs that they were added to, the risk outweighs the reward.

With respect to documentation, if you'd be so kind as to share the specific members that were causing you confusion, I'd like to see if we can't make that more clear.

**Comment by yuretz** [comment]
Hi @jsquire and thank you for your detailed response to my request. 

I understand your reasoning behind not yet supporting nullable reference type annotations, and while I still think it would have been a great thing for everybody, if done adequately of course, we will have to wait until you are willing and able to do it at the SDK level.

In regards to the problem we encountered, it was in the `Azure.Messaging.EventHubs.EventProcessorClient.ProcessErrorAsync` handler  where somehow we expected a provided `ProcessErrorEventArgs.PartitionId` property to be not null and therefore we failed to handle some of the processing errors correctly. It is not possible to provide any useful details from an original `ProcessErrorEventArgs`, because, as mentioned, it was a failure in the error handler itself, so these details were lost. 

At first it looked like a bug in the library itself, because `PartitionId` was declared as non-nullable, but then we discovered that nullable reference annotations were just not supported, which resulted in this feature request.
